app/pages/patients/patients.js

```diff
@@ -310,8 +311,9 @@ export let Patients = React.createClass({
   },
 
   componentWillReceiveProps: function(nextProps) {
-    let { loading, patients, invites, location, showingWelcomeMessage } = nextProps;
-    if (!loading && location.query.justLoggedIn) {
+    let { loading, loggedInUserId, patients, invites, location, showingWelcomeMessage } = nextProps;
+    
+    if (!loading && loggedInUserId && location.query.justLoggedIn) {
```

Comment:
So you had to add `loggedInUserId` here to catch when you log out and we clear the state then try to redirect you, is that it?